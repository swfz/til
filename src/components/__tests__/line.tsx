import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import Line from "../line"

describe("Line", () => {
  it("renders correctly", () => {
    const node = {
      frontmatter: {
        title: "hoge",
        tags: ["A", "B"],
        date: "2021-05-01",
      },
      fields: {
        slug: "/sample_hoge/fuga",
      },
    }
    const { getByText, getAllByLabelText } = render(<Line node={node}></Line>)

    // タイトルが表示されているか
    const title = getByText("hoge")
    expect(title).toBeInTheDocument()

    // タグが付いているか
    const tags = getAllByLabelText("tag")
    expect(tags.length).toBe(2)

    // 更新日が表示されているか
    const published = getByText(/2021-05-01/i)
    expect(published).toBeInTheDocument()
  })
})
