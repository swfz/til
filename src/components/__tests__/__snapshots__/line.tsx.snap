// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line snapshot 1`] = `
<div
  className="rounded border border-gray-100 p-6 shadow"
>
  <div
    className="text-xl"
  >
    <small>
      [
      2021-05-01
      ]
    </small>
    <h2
      className="pb-2 text-xl"
    >
      <a
        className="link"
        href="/sample_hoge/fuga"
      >
        hoge
      </a>
    </h2>
    <span
      className="flex flex-row flex-wrap gap-1"
    >
      <a
        aria-label="tag"
        className="tag"
        href="/tags/a/"
      >
        A
      </a>
      <a
        aria-label="tag"
        className="tag"
        href="/tags/b/"
      >
        B
      </a>
    </span>
  </div>
</div>
`;
